.github/workflows/soundness.yml

```diff
@@ -0,0 +1,19 @@
+name: Soundness
+
+on:
+  workflow_call:
+
+jobs:
+  swift-license-check:
+    name: Swift license headers check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Mark the workspace as safe
+        run: git config --global --add safe.directory ${GITHUB_WORKSPACE}
+      - name: Run license check
+        run: |
+          ./dev/license-check.sh
```

Comment:
Can we just inline this into `pull_request.yml`? I think it's a little easier to follow when they're grouped together.

Reply:
For scholars, we discussed this [here](https://github.com/grpc/grpc-swift-extras/pull/9#discussion_r1830811128) and I argued that having the checks grouped in the UI was worth the indirection which I think is required to get it.